JSDoc `@typedef` silently dropped when preceded by a multi-line `@property` function type and an interface-returning call

## Steps to reproduce

A top-level JSDoc `@typedef` in a checked `.js` file is silently **not registered** by `tsgo` when it is preceded by (1) a `@typedef` whose `@property` has a multi-line parenthesized function type, and (2) a `const` export initialized by a call returning a complex generic type (here `@endo/patterns`' `M.interface(...)`). The trailing `@typedef` disappears: every reference to it fails `TS2304`, and importers of the module get `TS2724`.

Minimal repro (depends on `@endo/patterns` — see reduction notes):

```js
// repro.js   (allowJs + checkJs)
import { M } from '@endo/patterns';

/**
 * @typedef {object} AppTransformer
 * @property {(
 *   app: number,
 *   targetHost: number,
 * ) => number} wrapApp
 */
export const AppTransformerI = M.interface('AppTransformer', {
  wrapApp: M.callWhen(M.await(M.remotable('x'))).rest(M.any()).returns(M.remotable('x')),
});

/**
 * @typedef {object} TargetRegistration
 * @property {() => Promise<void>} revoke
 */

/** @param {TargetRegistration} r */
export const use = r => r;
```

```jsonc
// tsconfig.json
{ "compilerOptions": { "module": "nodenext", "moduleResolution": "nodenext",
  "allowJs": true, "checkJs": true, "strict": false, "strictNullChecks": true,
  "noEmit": true, "skipLibCheck": true }, "include": ["repro.js"] }
```

Reduction notes:
- Replacing `M.interface`/`M.callWhen` with locally-defined functions (even with explicit generic signatures) makes the bug vanish, so resolving `@endo/patterns`' real generic types appears to be part of the trigger; I could not produce a zero-dependency repro.
- Removing the multi-line parenthesized `@property` function type from the *first* typedef also makes it vanish.
- A `@typedef` after an ordinary multi-line statement (e.g. a plain object literal) does **not** trigger it.

## Behavior with `typescript@6.0`

Clean — exit 0. `tsc` 6.0.3 binds the `@typedef` correctly; `TargetRegistration` resolves and is exported.

## Behavior with `tsgo`

`@typescript/native-preview` 7.0.0-dev.20260608.1:

```
repro.js(19,13): error TS2304: Cannot find name 'TargetRegistration'.
```

The typedef is never registered: references fail `TS2304: Cannot find name`, and importers of the module get `TS2724: ... has no exported member named 'TargetRegistration'`. In the real module that surfaced this, several `@typedef`s interleaved with `M.interface(...)` guards each silently vanished, cascading `TS2724` to ~7 importing files. Workaround: group all the `@typedef`s together so none is immediately adjacent to such a call.

--- a/internal/parser/parser.go
+++ b/internal/parser/parser.go
@@ -569,12 +569,12 @@ func (p *Parser) reparseTopLevelAwait(sourceFile *ast.SourceFile) *ast.Node {
 				p.nextToken()
 			}
 			if afterAwaitStatement < len(sourceFile.Statements.Nodes) {
-				nonAwaitStatement := sourceFile.Statements.Nodes[afterAwaitStatement]
-				if statement.End() == nonAwaitStatement.Pos() {
+				lastAwaitStatement := sourceFile.Statements.Nodes[afterAwaitStatement-1]
+				if statement.End() == lastAwaitStatement.End() {
 					// done reparsing this section
 					break
 				}
-				if statement.End() > nonAwaitStatement.Pos() {
+				if statement.End() > lastAwaitStatement.End() {
 					// we ate into the next statement, so we must continue reparsing the next span
 					i += 2
 					if i < len(p.possibleAwaitSpans) {

--- a/testdata/baselines/reference/compiler/topLevelAwaitReparseWithTypedef.symbols
+++ b/testdata/baselines/reference/compiler/topLevelAwaitReparseWithTypedef.symbols
@@ -1,0 +1,24 @@
+//// [tests/cases/compiler/topLevelAwaitReparseWithTypedef.ts] ////
+
+=== main.js ===
+// https://github.com/microsoft/typescript-go/issues/4252
+
+const obj = { await: 42 }
+>obj : Symbol(obj, Decl(main.js, 2, 5))
+>await : Symbol(await, Decl(main.js, 2, 13))
+
+export const x = obj.await;
+>x : Symbol(x, Decl(main.js, 3, 12))
+>obj.await : Symbol(await, Decl(main.js, 2, 13))
+>obj : Symbol(obj, Decl(main.js, 2, 5))
+>await : Symbol(await, Decl(main.js, 2, 13))
+
+/**
+ * @typedef {object} Foo
+ * @property {string} name
+ */
+
+/** @type {Foo} foo */
+let foo;
+>foo : Symbol(foo, Decl(main.js, 11, 3))
+

--- a/testdata/baselines/reference/compiler/topLevelAwaitReparseWithTypedef.types
+++ b/testdata/baselines/reference/compiler/topLevelAwaitReparseWithTypedef.types
@@ -1,0 +1,26 @@
+//// [tests/cases/compiler/topLevelAwaitReparseWithTypedef.ts] ////
+
+=== main.js ===
+// https://github.com/microsoft/typescript-go/issues/4252
+
+const obj = { await: 42 }
+>obj : { await: number; }
+>{ await: 42 } : { await: number; }
+>await : number
+>42 : 42
+
+export const x = obj.await;
+>x : number
+>obj.await : number
+>obj : { await: number; }
+>await : number
+
+/**
+ * @typedef {object} Foo
+ * @property {string} name
+ */
+
+/** @type {Foo} foo */
+let foo;
+>foo : Foo
+

--- a/testdata/tests/cases/compiler/topLevelAwaitReparseWithTypedef.ts
+++ b/testdata/tests/cases/compiler/topLevelAwaitReparseWithTypedef.ts
@@ -1,0 +1,16 @@
+// @noEmit: true
+// @checkJs: true
+// @filename: main.js
+
+// https://github.com/microsoft/typescript-go/issues/4252
+
+const obj = { await: 42 }
+export const x = obj.await;
+
+/**
+ * @typedef {object} Foo
+ * @property {string} name
+ */
+
+/** @type {Foo} foo */
+let foo;

